data/src/cache.js

```diff
@@ -0,0 +1,232 @@
+/**
+ * EJS_Cache
+ * Manages a cache of files using IndexedDB for storage.
+ */
+class EJS_Cache {
+    /**
+     * Creates an instance of EJS_Cache.
+     * @param {boolean} enabled - Whether caching is enabled.
+     * @param {EJS_STORAGE} storage - Instance of EJS_STORAGE for IndexedDB operations.
+     * @param {number} maxSizeMB - Maximum size of the cache in megabytes.
+     * @param {number} maxAgeMins - Maximum age of items (in minutes) before they are cleaned up.
+     */
+    constructor(enabled = true, storage, maxSizeMB = 4096, maxAgeMins = 7200) {
+        this.enabled = enabled;
+        this.storage = storage;
+        this.maxSizeMB = maxSizeMB;
+        this.maxAgeMins = maxAgeMins;
+        this.minAgeMins = Math.max(60, maxAgeMins * 0.1); // Minimum 1 hour, or 10% of max age
+
+        console.log('Initialized EJS_Cache with settings:', {
```

Comment:
Usually all log statements are only printed if debug is true. What do you think about this?